### ISSUE TITLE
gsudo stays waiting without -w option when launching Windows 'App Execution Aliases'

### ISSUE THREAD

**Issue by cawoodm** [gerardog/gsudo#46]
Launching `gsudo.exe wt.exe` should launch windows terminal and close gsudo. Unfortunately gsudo stays open (Windows 10).

![image](https://user-images.githubusercontent.com/498834/84587973-f3776d00-ae23-11ea-92ad-c413298cc571.png)

**Comment by gerardog** [comment]
Your `wt.exe` is located in `C:\Users\xxx\AppData\Local\Microsoft\WindowsApps\wt.exe` and it's 0 bytes long, right?.
It is a [windows execution alias](https://www.tiraniddo.dev/2019/09/overview-of-windows-execution-aliases.html), and since it is 0 bytes long, gsudo fails to detect if it is windows or a console app, (hence treating it as a console app and waiting for it to end)...

I will investigate if I can properly detect that the target is a windows execution alias...
In the meantime, you can easily workaround your problem using either `gsudo start wt.exe` or `gsudo -n wt.exe`.

**Comment by cawoodm** [comment]
Yep, that's it. Thanks muchly.

What a pity when you pin a `gsudo start wt.exe` shortcut to the taskbar it creates a new unrelated icon when you actually launch it. Is there no native way to start ANY app as Admin with a shortcut?

**Comment by cawoodm** [comment]
Note `gsudo -n wt.exe` did not work (error 0x..22) but `gsudo start wt.exe` did.

![image](https://user-images.githubusercontent.com/498834/85110189-c73c6180-b212-11ea-9afe-84daf4a86cab.png)

**Comment by gerardog** [comment]
Those new kind of aliases produce several issues, one is `gsudo` not being able to find out if the app is a windows or console app, other is the impossibility to launch as admin as reported in microsoft/terminal#2485.
I'll see if I can address those issues at gsudo without the `gsudo start wt` workaround.

If you want to create a shortcut to Windows Terminal now, you can try this powershell script:

``` powershell
$shortcutPath = [Environment]::GetFolderPath("Desktop") + "\Windows Terminal Elevated.lnk"

# Use this if installed via Scoop. 
$wtPath = "$home\scoop\apps\windows-terminal\current\WindowsTerminal.exe"

# Use this if installed via Chocolatey or Ms Store
$wtPath = (Get-Command 'wt.exe').Path
$gsudoPath = (Get-Command 'gsudo.exe').Path

$WshShell = New-Object -comObject WScript.Shell
$link = $WshShell.CreateShortcut($shortcutPath)
$link.TargetPath = $gsudoPath
$link.Arguments = "-n cmd /c cmd /c start $wtPath"
# alternative integrity level: $link.Arguments = "-n -i MediumPlus cmd /c cmd /c start $wtPath"

#Optional, set global Keyboard Hotkey
$link.Hotkey="Alt+Ctrl+T"

# Optional, download icon.
$icoPath = [IO.Path]::GetDirectoryName($wtPath) + "\wt.ico"
Invoke-RestMethod -Method Get -Uri "https://raw.githubusercontent.com/microsoft/terminal/master/res/terminal.ico" -OutFile $icoPath
$link.IconLocation="$icoPath,0"

$link.Save()
```

It will create a Windows Terminal Elevated shortcut on your desktop, map it "Alt+Ctrl+T", plus use the official icon.

**Comment by gerardog** [comment]
The fix for #65 seems to have also fixed this issue. Hurray. Was included in v0.7.3 release.
It's a workaround at this point and a more robust solution should be found, but nonetheless the problem is gone for Microsoft Store Apps.
